Description: Prefer ubuntu session as fallback over GNOME
 As GNOME, optionnally installed, is a Wayland session and ubuntu
 is now transitionned as a Xorg session, the fallback will prefer
 naturally the GNOME session. Ensure we prioritize the ubuntu
 session in fallback order.
Author: Didier Roche <didrocks@ubuntu.com>
Forwarded: not-needed

Index: gdm3-3.26.2.1/daemon/gdm-session.c
===================================================================
--- gdm3-3.26.2.1.orig/daemon/gdm-session.c
+++ gdm3-3.26.2.1/daemon/gdm-session.c
@@ -541,6 +541,14 @@ get_fallback_session_name (GdmSession *s
                 }
         }
 
+        name = g_strdup ("ubuntu");
+        if (get_session_command_for_name (self, name, NULL)) {
+                g_free (self->priv->fallback_session_name);
+                self->priv->fallback_session_name = name;
+                goto out;
+        }
+        g_free (name);
+
         name = g_strdup ("gnome");
         if (get_session_command_for_name (self, name, NULL)) {
                 g_free (self->priv->fallback_session_name);
